### PR TITLE
Cleanup for major version

### LINKED ISSUE #609: Use explicit API
## Feature Request

- Make the API as explicit as possible with typehints
- Close all classes
- Prefer `private` visibility

### PR DESCRIPTION
**Pull request by core23**
## Subject

Fixes #609

I am targeting this branch, because this is a BC break.

No changelog, because this removes all NEXT_MAJOR hacks which are already mentioned by the general deprecation note inside the UPRADE-4.0.md file.

### CODE REVIEW

File: tests/DependencyInjection/Compiler/TweakCompilerPassTest.php [hunk 1]
```diff
@@ -80,13 +80,19 @@ public function testProcessSameBlockId(): void
         $pass->process($this->container);
     }
 
+    public function blockIds(): \Generator
```
Comment (OskarStark) [inline tests/DependencyInjection/Compiler/TweakCompilerPassTest.php]:
```suggestion
    public function validBlockIds(): \Generator
```

?

**Comment by core23** [comment]
Ready for review! @sonata-project/contributors

File: src/Block/Service/MenuBlockService.php [hunk 2]
```diff
@@ -179,9 +181,10 @@ protected function getFormSettingsKeys(): array
      * Gets the menu to render.
      *
      *
+     *
```
Comment (phansys) [inline src/Block/Service/MenuBlockService.php]: These extra line LFs could be removed.

Comment (core23) [reply src/Block/Service/MenuBlockService.php]: Fixed

**Comment by core23** [comment]
Two approvements, so... Let's merge this thin 🙈

**Pull request merged**